dip-0003.md

```diff
@@ -77,6 +77,25 @@ The ProRegTx contains 3 public key IDs, which represent 3 different roles in the
 
 A single key can be used for multiple roles so the owner can also operate the masternode and/or issue proposal votes. Alternately, different keys can be used to delegate duties to other people. For example, owners might decide to use a masternode hosting service instead of self-hosting the masternode. To do so, they would use the operator key provided by the hosting service. The same applies to the voting key, which allows them to assign voting rights to someone else.
 
+The collateralOutpoint of the ProRegTx might refer to an external collateral or to an output which is part of the ProRegTx itself. If the hash of collateralOutpoint is null, it refers to
+an output of the ProRegTx. If it refers to an external collateral (hash of collateralOutpoint is not null), the collateral must exist and not be spent. The ProRegTx must also prove ownership
+of external collaterals by setting payloadSigSize and payloadSig to a valid signature which signs the following message:
```

Comment:
The message is signed by the collateral, correct?

Reply:
Yes, it ought to be signed as a "normal" message (with "magic prefix") using the private key associated with the collateral transaction.